#! /bin/sh /usr/share/dpatch/dpatch-run
## 05_saltbuffer-337116.dpatch by Marc Haber <mh+debian-packages@zugschlus.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix segfaults with -y option, #337116

@DPATCH@
diff -urNad trunk~/apg.c trunk/apg.c
--- trunk~/apg.c	2003-09-12 17:46:27.000000000 +0000
+++ trunk/apg.c	2006-10-17 10:07:54.000000000 +0000
@@ -662,7 +662,7 @@
 */
 char * crypt_passstring (const char *p)
 {
- char salt[10];
+ char salt[11];
  gen_rand_pass (salt, 10, 10, S_SL|S_CL|S_NB);
  return (crypt(p, salt));
 }
